### PR TITLE
fixes for a few coverity  issues

### PR DESCRIPTION
**Pull request by Paxxi**
Most of these should be harmless and fix the newest warnings that has popped up in coverity.

I'm not sure about cc_decoder.c though, it's been wrong for a long time so it may break something and maybe should be skipped?

### CODE REVIEW

**Comment by Paxxi** [comment]
@FernetMenta can you take a look at these? primarily cc_decoder

File: xbmc/guilib/XBTFReader.cpp [hunk 1]
```diff
@@ -31,16 +31,16 @@
 #include "PlatformDefs.h"
 
 #define READ_STR(str, size, file) \
-  if (!fread(str, size, 1, file)) \
+  if (1 != fread(str, size, 1, file)) \
```
Comment (fritsch) [inline xbmc/guilib/XBTFReader.cpp]: careful - this is wrong!

File: xbmc/network/upnp/UPnPServer.cpp [hunk 2]
```diff
@@ -278,9 +278,9 @@ CUPnPServer::Build(CFileItemPtr                  item,
             object->m_ObjectID = "0";
             object->m_ParentID = "-1";
             // root has 5 children
-            if (with_count) {
+            /*if (with_count) {
                 ((PLT_MediaContainer*)object)->m_ChildrenCount = 5;
-            }
+            }*/
```
Comment (fritsch) [inline xbmc/network/upnp/UPnPServer.cpp]: this one I don't understand

Comment (Montellese) [inline xbmc/network/upnp/UPnPServer.cpp]: `with_count` is forced to `false` a few lines up (see https://github.com/Paxxi/xbmc/blob/coverity/xbmc/network/upnp/UPnPServer.cpp#L265) so this is never true.

Comment (fritsch) [inline xbmc/network/upnp/UPnPServer.cpp]: Then - just removing that part of the code should be the way to go.

File: xbmc/guilib/XBTFReader.cpp (hunk 1 above)
Comment (Paxxi) [inline xbmc/guilib/XBTFReader.cpp]: It shouldnt be, docs says it returns count of elements read and count is always 1 so any return except 1 should be a failure

Comment (fritsch) [inline xbmc/guilib/XBTFReader.cpp]: Yes, you are right - it will do what we want in this situation, where we always want to read 1 element.

Comment (FernetMenta) [inline xbmc/guilib/XBTFReader.cpp]: I would remove this silly macro completely

**Comment by FernetMenta** [comment]
cc_decoder looks ok, thanks

File: xbmc/network/upnp/UPnPServer.cpp (hunk 2 above)
Comment (Paxxi) [inline xbmc/network/upnp/UPnPServer.cpp]: are you okay with just removing this part @Montellese ?

**Comment by Paxxi** [comment]
fixed upnpserver, left the code there but added comment to the hack above

jenkins build and merge please

**Pull request merged**